efm8load.py

```diff
@@ -271,30 +273,41 @@ def download(self, filename):
         #the bootloader protocol does not allow reading flash
         #however it allows to verify written bytes
         #we will exploit this feature to dump the flash contents
-        #for now assume 8kb flash
-        flash_size = 8 * 1024
+
         ih = IntelHex()
-        for address in range(flash_size):
-            #test one byte by byte
-            #first check 0x00
-            byte = 0
-            if (self.verify(address, [byte]) == RESPONSE.ACK):
-                ih[address] = byte
-            else:
-                #now start with 0xFF (empty flash)
-                for byte in range(0xFF, -1, -1):
-                    if (self.verify(address, [byte]) == RESPONSE.ACK):
-                        #success, the flash content on this address euals <byte>
-                        ih[address] = byte
-                        break
-            print("\r> flash[0x%04X] = 0x%02X" % (address, byte), end="")
-            sys.stdout.flush()
 
-        print("\n> finished")
+        # generate list with all possible hex adress values
+        byte_list = Counter()
+        for byte in range(0xFF, -1, -1):
+            byte_list[byte] += 1
+
+
+        for address in tqdm(
+            range(self.flash_size),
+            desc="Dumping Flash",
+            unit=" Bytes",
+            ascii=" #",
+            smoothing=0.01,
+            dynamic_ncols=True,
+            unit_scale=True,
+        ):  
+            sorted_byte_list = byte_list.most_common()
+            for curr_byte in sorted_byte_list:
+                byte=curr_byte[0]
+                if (self.verify(address, [byte]) == RESPONSE.ACK):
+                    #success, the flash content on this address equals <byte>
+                    ih[address] = byte
+                    byte_list[byte] += 1
+                    break
+            #print(" flash[0x%04X] = 0x%02X" % (address, byte))
+        print("\n> finished dumping")
 
-        #done, all flash contents have been read, now store this to the file
-        ih.write_hex_file(filename)
+        #done, all flash contents have been read, now store this to the files
+        ih.tofile("dumps/" + filename, format='hex')
+        ih.tofile("dumps/" + filename + ".bin", format='bin')
```

Comment:
i would not prepend dumps here, you can always pass dumps/file.hex as commandline param. e.g. i use it to dump to /tmp/dump.hex
adding bin is nice :)